src/main/scala/treadle/coverage/pass/AddCoverageExpressions.scala

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package treadle.coverage.pass
+
+import firrtl.Namespace
+import firrtl.PrimOps.Not
+import firrtl.ir.{Block, Circuit, Connect, DefModule, DoPrim, Expression, ExtModule, HasInfo, Module, Mux, NoInfo, Statement}
+import firrtl.passes.Pass
+import firrtl.stage.TransformManager.TransformDependency
+import treadle.coverage.{CoverageInfo, Ledger}
+
+import scala.collection.mutable
+
+/**
+  * Adds additional coverage statements to the Low FIRRTL source at every mux location.
+  * These statements can then be used to gather statement coverage information from inside of the Treadle tester
+  */
+object AddCoverageExpressions extends Pass {
+
+  override def prerequisites: Seq[TransformDependency] = firrtl.stage.Forms.LowForm
```

Comment:
You need to specify your invalidates.

Probably something similar to this:
https://github.com/freechipsproject/firrtl/blob/master/src/main/scala/firrtl/passes/RemoveAccesses.scala#L27

Reply:
You also invalidate `InferTypes` since the reference you create in your `Ledger` does not carry a type.